src/nomad_simulations/schema_packages/basis_set.py

```diff
@@ -0,0 +1,642 @@
+        """,
+    )
+
+    @property  # ? keep, or convert to `Quantity`?
+    def cutoff_radius(self) -> pint.Quantity:
+        """
+        Compute the cutoff radius for the plane-wave basis set, expressed in reciprocal coordinates.
+        """
+        if self.cutoff_energy is None:
+            return None
+        m_e = const.m_e * ureg(const.unit('electron mass'))
+        h = const.h * ureg(const.unit('Planck constant'))
+        return np.sqrt(2 * m_e * self.cutoff_energy) / h
+
+    def normalize(self, archive: EntryArchive, logger: BoundLogger) -> None:
+        super(BasisSet, self).normalize(archive, logger)
+        super(Mesh, self).normalize(archive, logger)
+
+
+class APWPlaneWaveBasisSet(PlaneWaveBasisSet):
+    """
+    A `PlaneWaveBasisSet` specialized to the APW use case.
+    Its main descriptors are defined in terms of the `MuffinTin` regions.
+    """
+
+    cutoff_fractional = Quantity(
+        type=np.float64,
+        shape=[],
+        description="""
+        The spherical cutoff parameter for the interstitial plane waves in the APW family.
+        This cutoff has no units, referring to the product of the smallest muffin-tin radius
+        and the length of the cutoff reciprocal vector ($r_{MT} * |K_{cut}|$).
+        """,
+    )
+
+    def set_cutoff_fractional(
+        self, mt_r_min: pint.Quantity, logger: BoundLogger
+    ) -> None:
+        """
+        Compute the fractional cutoff parameter for the interstitial plane waves in the LAPW family.
+        This parameter is defined wrt the smallest muffin-tin region.
+        """
+        reference_unit = 'angstrom'
+        if self.cutoff_fractional is not None:
+            logger.info(
+                '`APWPlaneWaveBasisSet.cutoff_fractional` already defined. Will not overwrite.'
+            )  #! extend implementation
+            return
+        elif self.cutoff_energy is None or mt_r_min is None:
+            logger.warning(
+                '`APWPlaneWaveBasisSet.cutoff_energy` and `APWPlaneWaveBasisSet.radius` must both be defined. Aborting normalization step.'
+            )
+            return
+        self.cutoff_fractional = self.cutoff_radius.to(
+            f'1 / {reference_unit}'
+        ) * mt_r_min.to(reference_unit)
+
+
+class AtomCenteredFunction(ArchiveSection):
+    """
+    Specifies a single function (term) in an atom-centered basis set.
+    """
+
+    pass
+
+    # TODO: design system for writing basis functions like gaussian or slater orbitals
+
+
+class AtomCenteredBasisSet(BasisSet):
+    """
+    Defines an atom-centered basis set.
+    """
+
+    functional_composition = SubSection(
+        sub_section=AtomCenteredFunction.m_def, repeats=True
+    )  # TODO change name
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+        # self.name = self.m_def.name
+        # TODO: set name based on basis functions
+        # ? use naming BSE
```

Comment:
It might conflict with elasticsearch and BetheSalpeterEquation, so we have to carefully check this

Reply:
> # ? use naming BSE

This refers to the formatting in the Basis Set Exchange, not changing the name to BSE. I've modified the comment for clarity.